impl/src/main/java/org/pentaho/di/trans/dataservice/streaming/execution/StreamingGeneratedTransExecution.java

```diff
@@ -226,4 +254,17 @@ public void clearRowConsumer( final Observer<RowMetaAndData> consumer ) {
   protected PublishSubject<List<RowMetaAndData>> getGeneratedDataObservable() {
     return generatedDataObservable;
   }
+
+  /**
+   * Removes all the consumers that have the flag to keep sending set to false
+   */
+  private void removeNotActiveConsumer() {
```

Comment:
we should be providing consumers with a disposable that handles the cleanup, instead of checking the AtomicBoolean to do it